### PR TITLE
Attack Detector - Reactive FP Mitigation

### PR DESCRIPTION
**Pull request by Vxatz**
- Keeping the same logic as Scam Detector's but querying for more days each time (2, 7, 30 89 days)

### CODE REVIEW

**Comment by christian-forta** [comment]
@Vxatz, could you adjust to merge into the next train: https://github.com/forta-network/starter-kits/tree/AD_3_52_MEV_Consensus

File: alert-combiner-py/src/blockchain_indexer_service.py [hunk 1]
```diff
@@ -38,6 +39,25 @@ def get_etherscan_url(chain_id):
             return "https://api.snowtrace.io"
 
         raise Exception("Chain ID not supported")
+    
+    @staticmethod
+    def get_first_block_number(chain_id):
```
Comment (christian-forta) [inline alert-combiner-py/src/blockchain_indexer_service.py:44]: is this going to pose a problem as time progresses? Should these offsets from the current block number and get updated regularly?

File: alert-combiner-py/src/findings.py [hunk 2]
```diff
@@ -111,18 +111,22 @@ def create_finding(block_chain_indexer, addresses: str, victim_address: str, vic
                        })
 
     @staticmethod
-    def alert_FP(address: str, label: str, metadata: dict) -> Finding:
+    def alert_FP(address: str, label: str, metadata_list: list) -> Finding:
```
Comment (christian-forta) [inline alert-combiner-py/src/findings.py:114]: dont quite understand why we now have a metadata list; i see above you create a list, but only pass in one metadata obj?

**Review by christian-forta — APPROVED** [review]

File: alert-combiner-py/src/findings.py (hunk 2 above)
Comment (Vxatz) [reply alert-combiner-py/src/findings.py:114]: @christian-forta, there may be more than one labels due to different label metadata. In the code I'm doing

> if entity not in REACTIVE_LIKELY_FPS:
                        REACTIVE_LIKELY_FPS[entity] = {'label': label_name, 'metadata': [metadata]}
                    else:
                        REACTIVE_LIKELY_FPS[entity]['metadata'].append(metadata)
                        
and then passing

`findings.append(AlertCombinerFinding.alert_FP(address, REACTIVE_LIKELY_FPS[address]['label'], REACTIVE_LIKELY_FPS[address]['metadata']))`

**Comment by Vxatz** [comment]
@christian-forta , I changed the base branch to 3.52

File: alert-combiner-py/src/blockchain_indexer_service.py (hunk 1 above)
Comment (Vxatz) [reply alert-combiner-py/src/blockchain_indexer_service.py:43]: By the way, I noticed that the endBlockNumber is smaller than Arbitrum's latest, so I changed that (by adding an extra 9 to the 999...99)

File: alert-combiner-py/src/constants.py [hunk 3]
```diff
@@ -31,6 +34,7 @@
                       ("0xa91a31df513afff32b9d85a2c2b7e786fdd681b3cdd8d93d6074943ba31ae400", "FUNDING-TORNADO-CASH-HIGH"),
                       ("0xd6e19ec6dc98b13ebb5ec24742510845779d9caf439cadec9a5533f8394d435f", "POSITIVE-REPUTATION-1"),
                       ("0xd6e19ec6dc98b13ebb5ec24742510845779d9caf439cadec9a5533f8394d435f", "POSITIVE-REPUTATION-2"),
+                      ("0xd6e19ec6dc98b13ebb5ec24742510845779d9caf439cadec9a5533f8394d435f", "POSITIVE-REPUTATION-3"),
```
Comment (christian-forta) [inline alert-combiner-py/src/constants.py:37]: thx :)

**Review by christian-forta — APPROVED** [review]
makes the code cleaner too. thanks Vasilis.

**Pull request merged**